packages/fuel-indexer-macros/src/schema.rs

```diff
@@ -350,22 +361,75 @@ fn process_type_def(
                     #to_row
                     #decoder
                 };
+
+                let unwrap_or_def = if field_type.nullable {
+                    if EXTERNAL_FIELD_TYPES.contains(field_typ_name.as_str()) {
+                        unwrap_or_default_for_external_type(field_typ_name.clone())
+                    } else {
+                        quote! { .unwrap_or_default() }
+                    }
+                } else {
+                    quote! {}
+                };
+
+                let to_bytes = if EXTERNAL_FIELD_TYPES.contains(field_typ_name.as_str()) {
+                    to_bytes_method_for_external_type(field_typ_name.clone())
+                } else if !ASREF_BYTE_TYPES.contains(field_typ_name.as_str()) {
+                    quote! { .to_le_bytes() }
+                } else {
+                    quote! {}
+                };
+
+                if field_set.contains(&"id".to_string())
+                    && !INTERNAL_INDEXER_ENTITIES.contains(object_name.as_str())
```

Comment:
Same comment here with regard to the magic `"id"` string. Let's use `IdCol` if possible.